helper/resource/testing_new_import_state.go

```diff
@@ -458,8 +458,12 @@ func appendImportBlockWithIdentity(config string, resourceName string, identityV
 
 		case string:
 			configBuilder += fmt.Sprintf(`		%q = %q`+"\n", k, v)
-
 		default:
+			// It's valid for identity attributes to be null, we can just omit it from config
+			if v == nil {
+				break
+			}
```

Comment:
Testing shows that this should be enough for checking `nil` in this exact use-case. Since the only producer of this data is `terraform-json`, feels like we won't need to also use reflection? 🤔

Reply:
Nice! Suggestion: it's safer (and/or less cognitive overhead) to check for `nil` before the `switch`, since `nil` values can have a type.